amlb/framework_definitions.py

```diff
@@ -0,0 +1,182 @@
+import copy
+import itertools
+import logging
+from typing import Union, List
+
+from amlb.utils import Namespace, config_load
+
+log = logging.getLogger(__name__)
+
+
+def load_framework_definitions(frameworks_file: Union[str, List[str]], resource) -> Namespace:
+    """ Load the framework definition listed in the framework file(s).
+
+    Loads the definition(s) from the file(s),
+    :param frameworks_file:
+    :return: Namespace containing each framework definition,
+    """
+    frameworks = load_and_merge_framework_definitions(frameworks_file)
+    sanitize_and_add_defaults(frameworks, resource)
+    log.debug("Available framework definitions:\n%s", frameworks)
+    return frameworks
+
+
+def load_and_merge_framework_definitions(frameworks_file: Union[str, List[str]]) -> Namespace:
+    """ Load and merge the framework file(s), does not allow duplicate definitions. """
+    log.info("Loading frameworks definitions from %s.", frameworks_file)
+    if not isinstance(frameworks_file, list):
+        frameworks_file = [frameworks_file]
+
+    definitions_by_file = [config_load(file) for file in frameworks_file]
+    for d1, d2 in itertools.combinations([set(dir(d)) for d in definitions_by_file], 2):
+        if d1.intersection(d2) != set():
+            raise ValueError(f"Duplicate entry '{d1.intersection(d2).pop()}' found.")
```

Comment:
It should find duplicate framework definitions, if e.g. a framework is defined in two different files (if it is in the same file, the `yaml` parser would catch it).  I guess it's a matter of preference. At the very least I think it makes sure that we do not allow this silently. If we do find a double definition, I think a warning is reasonable. Otherwise the user might accidentally run the framework as it was defined in a different file. 

I think raising an error instead is also reasonable, as the user can still define their own version of an existing framework either from scratch (with a different name), through extensions (idem), or by only providing one definition (and omitting the other).

Reply:
oh right!  so you're just verifying that there's not 2 definitions with the same name, makes sense.
Now, warning vs error?
I would tend to prefer warning here, because the custom definition should take precedence over the default one (at least that's how it used to work), so I don't see any reason to strictly forbid overriding as soon as the framework is then selected in a predictible way.

Reply:
What if two non-default files have custom definitions with the same identifier?